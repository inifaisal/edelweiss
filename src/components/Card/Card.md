Card:

```js
<div class='doc'>
  <Card>
    <p>Content</p>
  </Card>
  <Card elevation={3}>
    <p>Content</p>
  </Card>
  <Card elevation={4}>
    <p>Content</p>
  </Card>

  <Card elevation={5}>
    <p>Content</p>
  </Card>
</div>
```
