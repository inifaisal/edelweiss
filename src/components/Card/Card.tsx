import React from 'react';
import Classnames from 'classnames';

export type CardProps = {
  elevation?: number;
  children: any;
};

const Card = ({ elevation, children }: CardProps) => {
  const cardCls = Classnames('card', elevation && `elevation-${elevation}`);

  return <div className={cardCls}>{children}</div>;
};

export default Card;
