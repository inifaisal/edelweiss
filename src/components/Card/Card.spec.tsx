import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

it('renders without crashing', () => {
  const actual = shallow(
    <Card>
      <p>card contents</p>
    </Card>
  );

  expect(actual).toMatchSnapshot();
});
