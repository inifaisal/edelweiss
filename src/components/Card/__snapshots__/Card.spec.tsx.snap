// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card>
    <p>
      card contents
    </p>
  </Card>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        card contents
      </p>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "card contents",
      },
      "ref": null,
      "rendered": "card contents",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          card contents
        </p>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "card contents",
        },
        "ref": null,
        "rendered": "card contents",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
